templates/docs/examples/patterns/navigation/_script-sliding.js

```diff
@@ -190,6 +206,22 @@ const initNavigationSliding = () => {
       document.addEventListener('keyup', keyPressHandler);
     };
   }
+
+  // throttle util (for window resize event)
+  var throttle = function (fn, delay) {
```

Comment:
We could, but Vanilla not being a JS library, we don't really have a JS dependency, or a shared JS files. We also have to make sure that all that is needed for certain example gets included (rendered) in the example HTML.

So far, we've been coping these, but I agree that we are getting to a point where it would be worth having a look at organising the Vanilla JS a bit better.

Reply:
Why isn't it part of browser native JS api yet? 🤷‍♂️